dali/base/dasds.cpp

```diff
@@ -5753,52 +5767,52 @@ class CStoreHelper : implements IStoreHelper, public CInterface
-            unsigned xmlCrc{0};
-            try
-            {
-                xmlCrc = saveStoreToFile(root, iXmlFileIOTmpStore, StoreFormat::XML);
-            }
-            catch (IException *e)
-            {
-                StringBuffer errMsg;
-                EXCLOG(e, errMsg.append("Failed to save XML temporary store to : ").append(temporaryXmlFileSaveName).str());
-                e->Release();
-                iXmlFileTmpStore->remove();
-                return;
-            }
-
-            // Save store as binary
             OwnedIFile iBinaryFileTmpStore;
+            unsigned xmlCrc{0};
             unsigned binaryCrc{0};
-            unsigned *binaryCrcPtr{nullptr};
+
+            // Execute save operations
             if (saveBinary)
             {
-                StringBuffer tmpStoreName;
-                constexpr const char *temporaryBinaryFileSaveName = "dali_store_tmp.bin";
-                OwnedIFileIO iBinaryFileIOTmpStore = createUniqueFile(location, temporaryBinaryFileSaveName, NULL, tmpStoreName);
-                iBinaryFileTmpStore.setown(createIFile(tmpStoreName));
+                std::future<IFile *> xmlStoreSavedFuture;
+                if (saveAsync)
+                {
+                    // Save the XML in the background
+                    auto asyncXmlSaveFunc = [this, &xmlCrc, root]() -> IFile *
+                    {
+                        LOG(MCdebugProgress, "Saving XML store asynch");
+                        return saveStoreToFile(root, StoreFormat::XML, xmlCrc);
+                    };
+                    xmlStoreSavedFuture = std::async(std::launch::async, asyncXmlSaveFunc);
+                }
+                else // if synchronous save XML 1st
+                    iXmlFileTmpStore.setown(saveStoreToFile(root, StoreFormat::XML, xmlCrc));
+
                 try
                 {
-                    binaryCrc = saveStoreToFile(root, iBinaryFileIOTmpStore, StoreFormat::BINARY);
-                    binaryCrcPtr = &binaryCrc;
+                    iBinaryFileTmpStore.setown(saveStoreToFile(root, StoreFormat::BINARY, binaryCrc));
                 }
                 catch (IException *e)
                 {
-                    StringBuffer errMsg;
-                    EXCLOG(e, errMsg.append("Failed to save binary temporary store to : ").append(temporaryBinaryFileSaveName).str());
+                    OWARNLOG(e, "Exception - Error saving Binary store");
                     e->Release();
-                    iBinaryFileTmpStore->remove();
+                    // Any binary save failure is ignored as only the XML save is critical
                 }
+                catch (DALI_CATCHALL)
+                {
+                    OWARNLOG("Unknown exception - Error saving Binary store");
+                    // Any binary save failure is ignored as only the XML save is critical
+                }
+
+                if (saveAsync)
```

Comment:
future: it would probably have been worth commenting that if asyncXmlSaveFunc failed (threw an exception) then this future will result in an exception